nestkernel/communicator.cpp

```diff
@@ -146,7 +149,9 @@ nest::Communicator::init( int* argc, char** argv[] )
   // generate and commit struct
   MPI_Type_struct( 2, blockcounts, offsets, source_types, &MPI_OFFGRID_SPIKE );
   MPI_Type_commit( &MPI_OFFGRID_SPIKE );
-
+  // set up order of communication if using CPEX
+  if ( !use_Allgather_ )
+    init_communication();
   initialized_ = true;
```

Comment:
The CPEX communication was removed in #24. This seems like a relict of old times coming back. There are many parts in the Communicator class that are affected by this. One should make a diff of the Communicator before CPEX removal and this version and see, what actually needs changing. As far as I can see, only these functions need to be added:

    static void communicate( std::vector< ulong_t >& send_buffer,
        std::vector< ulong_t >& recv_buffer,
        std::vector< int >& displacements );
    static void communicate( std::vector< int_t >& send_buffer,
        std::vector< int_t >& recv_buffer,
        std::vector< int >& displacements );
and

    /*
     * Sum across all rank
     */
    static void communicate_Allreduce_sum_in_place( double_t buffer );
    static void communicate_Allreduce_sum_in_place( std::vector< double_t >& buffer );
    static void communicate_Allreduce_sum_in_place( std::vector< int_t >& buffer );
    static void communicate_Allreduce_sum( std::vector< double_t >& send_buffer,
      std::vector< double_t >& recv_buffer );